File: src/main/java/com/jose/mementoweb/dto/UpdateArticleRequest.java
package com.jose.mementoweb.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record UpdateArticleRequest(
    @NotBlank
    @Size(max = 255)
    String title,

    @Size(max = 255)
    String pretitle,

    String excerpt,
    String body
) {
}

File: frontend/src/app/admin/articles/update-article-request.ts
export interface UpdateArticleRequest {
  title: string;
  pretitle: string | null;
  excerpt: string | null;
  body: string | null;
  coverImageUrl: string | null;
  coverImageAlt: string | null;
  recommendedAudioTitle: string | null;
  recommendedAudioAuthor: string | null;
  recommendedAudioUrl: string | null;
}
